snippets/csharp/System.Xml/XmlDocument/Overview/xmlhelpermethods.cs

```diff
@@ -98,35 +98,24 @@ public XmlDocument LoadDocumentWithSchemaValidation(bool generateXML, bool gener
             // Helper method to retrieve schema.
             XmlSchema schema = getSchema(generateSchema);
 
-            if (schema == null)
```

Comment:
Is it OK to remove this validation here? Adding into schema set below would crash on null schema.

Reply:
Yes but please add test cases showing old behavior being broken and then fixing the testcase. This code is very old and touching it is always a risk of breaking someone

Reply:
The original code did not have broken behavior. It was poorly written and a reader won't know that it is 22 year old. A reader is always looking for the best practices way to write C#. The origin code is poorly written because ignores the generateXML parameter if the XML file was found on disk and it ignores the generateSchema parameter if the schema file was found on disk. Ignoring the parameters passed to a method on success is not proper behavior for well written code. @MSDN-WhiteKnight  and came up with an example where the XML is only read from file if generateXML=false. The schema is only read from a file if generateSchema=false. The parameters are no longer ignored. generateSchema is set to true to work around the XML file not found error by generating the XML programmatically. generateSchema is set to true to work around the schema file not found error by generating the schema programmatically. 

The other changes already committed for this page were to remove the Windows Forms specific code and to add a method to the snippet that cause the code not to compile when pasted into a file.

The method signatures in the change remained the same. The new code just more appropriately uses the exception to handle an error and the new code uses all the parameters passed to the methods (they are not ignored).

The code may be old but we can still clean it up.